src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -1784,55 +1784,70 @@ public bool IsOnTypeHeader(SyntaxNode root, int position)
                 return false;
             }
 
-            var start = GetStartOfNodeExcludingAttributes(typeDecl);
-            var end = typeDecl.GetTypeParameterList()?.GetLastToken().FullSpan.End ??
-                        typeDecl.Identifier.FullSpan.End;
-
-            return position >= start && position <= end;
+            return IsOnHeader(position, typeDecl, typeDecl.Identifier);
         }
 
-        public bool IsInPropertyDeclarationHeader(SyntaxNode node)
+        public bool IsOnPropertyDeclarationHeader(SyntaxNode root, int position, out SyntaxNode propertyDeclaration)
         {
-            var propertyDeclaration = node.GetAncestorOrThis<PropertyDeclarationSyntax>();
+            var node = TryGetAncestorForLocation<PropertyDeclarationSyntax>(position, root);
```

Comment:
the 'node' temporary seems superfluous (this comment applies to all your helpers).

Reply:
I couldn't find a way to make it work without it. The out parameter needs to be SyntaxNode (out params aren't "covariant" (I suppose they can't be as out params aren't CLR concept) so if the Interface says `out SyntaxNode` it needs to be of type SyntaxNode. And I'd really like to keep the reasonable name for the out parameter.

I also need a concrete type version of the variable in method for operations. Sure, I could always cast but that doesn't feel "more readable". I could move things around but I'm not sure if any other permutation would be better.

Reply:
You're right. I'm ok with this. Thanks!